components/org.wso2.carbon.security.caas/src/main/java/org/wso2/carbon/security/caas/user/core/store/AuthorizationStore.java

```diff
@@ -407,34 +431,42 @@ public void deletePermission(Permission permission) throws AuthorizationStoreExc
                 .get(permission.getAuthorizationStoreId());
 
         if (authorizationStoreConnector == null) {
-            throw new StoreException("Invalid authorization store id.");
+            throw new StoreException(String.format("No authorization store found for the given id %s",
+                    permission.getAuthorizationStoreId()));
         }
 
         authorizationStoreConnector.deletePermission(permission.getPermissionId());
     }
 
     /**
      * Add a new Role list by <b>replacing</b> the existing Role list. (PUT)
+     * Sending a null or empty list will remove all of the roles associated with the specified user in all available
+     * authorization stores.
      * @param userId Id of the user.
      * @param identityStoreId Identity store id of the user.
      * @param newRoleList List of Roles needs to be assigned to this User.
      * @throws AuthorizationStoreException Authorization store exception.
-     * @throws IdentityStoreException Identity Store Exception.
      */
     public void updateRolesInUser(String userId, String identityStoreId, List<Role> newRoleList)
-            throws AuthorizationStoreException, IdentityStoreException {
+            throws AuthorizationStoreException {
+
+        if (newRoleList == null || newRoleList.isEmpty()) {
+            for (AuthorizationStoreConnector authorizationStoreConnector : authorizationStoreConnectors.values()) {
+                authorizationStoreConnector.updateRolesInUser(userId, identityStoreId, newRoleList);
```

Comment:
For the same identity store (JDBC), is this handled with multiple transactions ?